src/client/dfs/dfs.c

```diff
@@ -2326,6 +2349,9 @@ dfs_open(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 	else if (!S_ISDIR(parent->mode))
 		return ENOTDIR;
 
+	if (stbuf && !(flags & O_CREAT))
+		return ENOTSUP;
+
```

Comment:
Update: Actually, I see we get the stat info from `entry` below here. So is there some other reason why we can't support this?

Reply:
We possibly could for open_file, I'm working on adding test coverage for this currently, particularly where O_CREAT is set but not O_EXCL and the file already exists.

open_dir() works with O_CREAT as expected, but the rest of open_dir() is only called for / by dfs_mount() so could probably just be a seperate function.

open_symlink() doesn't have any code to support the no O_CREAT path at all.

At the minute this is an internal function, and only called from dfuse, and always with O_CREAT, but it's also an odd function in that it's open, but specifies *what* it's trying to open, it should probably be dfs_create() rather than dfs_open() or dfs_open_stat() given what it does, but this is still an active area of development, and why it's best not to commit to interfaces too early in the development cycle IMO.

In short, it's not that we can't support this, it's that we don't need to for current use-cases and are still working through getting the API correct.

Reply:
I guess I don't object to that 👍 
Really I was thinking that if we _**did**_ have the stat info available for all cases, then we could reduce complexity here by removing this check, and setting stat info if and only if `stbuf` was passed in - similar to dfs_lookup, etc.
But you're right, this is just internal, so we can do whatever we want with it.

Reply:
If we did have it, then we should use it and not discard it, but we neither have it in this case, or have a code-path that could get to this case.